packages/core/src/components/document-component/document-component-heading.tsx

```diff
@@ -1,24 +1,44 @@
 import { h, FunctionalComponent } from "@stencil/core";
 import { JSXBase } from "@stencil/core/internal";
+import { DocumentComponentMode } from "./document-component.models";
 
 interface DocumentComponentHeadingProps {
   heading: "h2" | "h3" | "h4" | "h5" | "h6";
+  mode: DocumentComponentMode;
+  href?: string;
 }
 
 export const Heading: FunctionalComponent<
   DocumentComponentHeadingProps & JSXBase.HTMLAttributes<HTMLHeadingElement>
 > = ({ heading, ...props }, children) => {
+  let headingElement: HTMLHeadingElement;
```

Comment:
Het type waar je naar verwijst is dat van https://developer.mozilla.org/en-US/docs/Web/API/HTMLHeadingElement. Maar verderop zet je JSX.

Volgens mij is het niet mogelijk om kindjes te beperken in JSX. Met dit type is `headingElement = <p />` nog steeds mogelijk.

Ik denk dat je met `VNode` uit `@stencil/core` wilt werken. Dat is tevens één van de return types van een Stencil Functional Component (return type is `VNode | VNode[]`).